cmd/vicadmin/vicadm.go

```diff
@@ -544,27 +575,26 @@ func (s *server) serve() error {
 	s.handleFunc("/container-logs.tar.gz", s.tarContainerLogs)
```

Comment:
I had mentioned this to Ian in person, we could implement the http.Handler interface to achieve this.  There are probably better ways too, but agreed it would be best to authenticate all routes in a single place.  I think its also ok to do that refactoring in this PR or a later PR.